lms/djangoapps/branding/__init__.py

```diff
@@ -17,14 +17,26 @@
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 
 
-def get_visible_courses():
+def get_visible_courses(org=None):
     """
-    Return the set of CourseDescriptors that should be visible in this branded instance
+    Return the set of CourseOverviews that should be visible in this branded instance
     """
-    filtered_by_org = microsite.get_value('course_org_filter')
-    courses = CourseOverview.get_all_courses(org=filtered_by_org)
+    microsite_org = microsite.get_value('course_org_filter')
+
+    if org and microsite_org:
+        # When called in the context of a microsite, return an empty result if the org
+        # passed by the caller does not match the designated microsite org.
+        courses = CourseOverview.get_all_courses(org=org) if org == microsite_org else []
+    else:
+        target_org = org or microsite_org
```

Comment:
The code only makes it to this point if one of `org` or `microsite_org` is defined. If both `org` and `microsite_org` were defined, the code would have fallen into the first branch of the conditional, wherein an equality check is performed. See the comment on lines 27-28 for an explanation of the equality check.